### PR TITLE
[AKS] `az aks create` and `az aks nodepool add`: Add crg-id option to create nodepool with Capacity Reservation Group

### PR DESCRIPTION
**Pull request by tonychen15**
**Related command**
az aks create --crg-id
az aks nodepool add --crg-id

**Description**
We want to move AKs Capacit Reservation Group create feature from preview to stable release.
An option --crg-id is added to the above two commands.
With this command, customers can create nodepool with CRG.

**Testing Guide**
Please refer to test_aks_commands.py

**History Notes**
[Component Name 1] BREAKING CHANGE: `az command a`: Make some customer-facing breaking change
[Component Name 2] `az command b`: Add some customer-facing feature

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Thank you for your contribution! We will review the pull request and get back to you soon.

**Review by FumingZhang — APPROVED** [review]
LGTM

File: src/azure-cli/azure/cli/command_modules/acs/tests/latest/test_aks_commands.py [hunk 1]
```diff
@@ -9542,4 +9542,60 @@ def test_aks_nodepool_update_with_nsg_control(self, resource_group, resource_gro
         cmd = 'aks delete --resource-group={resource_group} --name={name} --yes --no-wait'
         self.cmd(cmd, checks=[
             self.is_empty(),
-        ])    
\ No newline at end of file
+        ])
+
+    @AllowLargeResponse()
+    @AKSCustomResourceGroupPreparer(random_name_length=17, name_prefix='clitest', location='westus2')
```
[the remark this answers is not in the document]
Comment (tonychen15) [reply src/azure-cli/azure/cli/command_modules/acs/tests/latest/test_aks_commands.py:9548]: Agree with Fuming. It seems the pipeline live test is case sensitive compared to local live test. The CRG resource id which Porrtal shows is " /subscriptions/26fe00f8-9173-4872-9134-bb1d2e00343a/resourceGroups/STAGING-CRG-RG/providers/Microsoft.Compute/capacityReservationGroups/crg-1", and the test case is using "/subscriptions/26fe00f8-9173-4872-9134-bb1d2e00343a/resourceGroups/staging-crg-rg/providers/Microsoft.Compute/capacityReservationGroups/crg-1". The difference is the case of resource group name. This should be caused by the test environment issue, but not CLI itself.

File: src/azure-cli/azure/cli/command_modules/acs/_help.py [hunk 2]
```diff
@@ -518,6 +518,9 @@
   - name: --nodepool-asg-ids
     type: string
     short-summary: The IDs of the application security groups to which the node pool's network interface should belong. When specified, format should be a space-separated list of IDs.
+  - name: --crg-id
+    type: string
+    short-summary: The crg-id used to associate the new cluster with the existed Capacity Reservation Group resource.
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/acs/_help.py:523]:
```suggestion
    short-summary: The crg id used to associate the new cluster with the existed Capacity Reservation Group resource.
```

**Comment by zhoxing-ms** [comment]
> BREAKING CHANGE

May I ask what's the breaking change this PR has?

File: src/azure-cli/azure/cli/command_modules/acs/_help.py [hunk 3]
```diff
@@ -1528,6 +1533,10 @@
   - name: --asg-ids
     type: string
     short-summary: The IDs of the application security groups to which the node pool's network interface should belong. When specified, format should be a space-separated list of IDs.
+  - name: --crg-id
+    type: string
+    short-summary: The crg-id used to associate the new nodepool with the existed Capacity Reservation Group resource.
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/acs/_help.py:1538]:
```suggestion
    short-summary: The crg id used to associate the new nodepool with the existed Capacity Reservation Group resource.
```

**Comment by yanzhudd** [comment]
Please let me confirm with you if this is a breaking change, since I noticed that the change is only adding a param which is not required for input.

**Comment by tonychen15** [comment]
> Please let me confirm with you if this is a breaking change, since I noticed that the change is only adding a param which is not required for input.

Yeah, you're right. We planned to add a new parameter but later we didn't add it.

**Comment by tonychen15** [comment]
I quickly checked that "Test Extensions Loading Python311 failed" error, I don't think changing resource group name case in test file may lead to this error. I merged the origin/dev branch to resolve some conflicts, and those conflicts should not lead to that error either.

**Comment by yanzhudd** [comment]
> I quickly checked that "Test Extensions Loading Python311 failed" error, I don't think changing resource group name case in test file may lead to this error. I merged the origin/dev branch to resolve some conflicts, and those conflicts should not lead to that error either.

The CI has been resolved.
Please address the comments raised by zhoxing-ms above.

**Comment by tonychen15** [comment]
I quickly checked that "Test Extensions Loading Python311 failed" error, I don't think changing resource group name case in test file may lead to this error. I merged the origin/dev branch to resolve some conflicts, and those conflicts should not lead to that error neither. 

> > I quickly checked that "Test Extensions Loading Python311 failed" error, I don't think changing resource group name case in test file may lead to this error. I merged the origin/dev branch to resolve some conflicts, and those conflicts should not lead to that error either.
> 
> The CI has been resolved. Please address the comments raised by zhoxing-ms above.

I have applied Zhoxing's suggestion.

**Comment by tonychen15** [comment]
@yanzhudd @zhoxing-ms Do you have any concern to merge this PR? If not, could you please merge it? Thanks.

**Review by yanzhudd — APPROVED** [review]

**Pull request merged**